tests/src/JIT/opt/FastTailCall/GitHubIssue12479.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+public class GitHubIssue12479
+{
+    public static int callee(int one,
+                             int two,
+                             int three,
+                             int four,
+                             int five,
+                             int six,
+                             int seven,
+                             int eight)
+    {
+        int count = 0;
+
+        // Make sure this is not inlined.
+        for (int i = 0; i < one; ++i)
+        {
+            if (i % 4 == 0) ++count;
+        }
+
+        return count;
+    }
+
+    // Eight incoming arguments, all passed in registers
+    //
```

Comment:
You might want to specify this is specifically for Linux/x64.

Possibly this should be combined with the FastTailCallCandidates.cs tests, since it looks so similar.